conan/tools/cmake/toolchain/blocks.py

```diff
@@ -1003,6 +1003,46 @@ def context(self):
                 "winsdk_version": winsdk_version,
                 "gen_platform_sdk_version": gen_platform_sdk_version}
 
+class ExtraVariablesBlock(Block):
+    template = textwrap.dedent(r"""
+        {% if extra_variables %}
+        {% for key, value in extra_variables.items() %}
+        set({{ key }} {{ value }})
+        {% endfor %}
+        {% endif %}
+    """)
+
+    CMAKE_CACHE_TYPES = ["BOOL","FILEPATH", "PATH", "STRING", "INTERNAL"]
+
+    def get_exact_type(self, key, value):
+        if type(value) == str:
+            return f"\"{value}\""
+        elif type(value) == int or type(value) == float:
+            return value
+        elif type(value) == dict:
+            var_value = self.get_exact_type(key, value.get("value"))
+            is_cache = value.get("cache")
+            if is_cache == True or str(is_cache).lower() == 'true':
+                var_type = value.get("type")
+                if not var_type:
+                    raise ConanException(f'CMakeToolchain needs type defined for cache variable "{key}"')
+                if var_type not in self.CMAKE_CACHE_TYPES:
+                    possible_types = ', '.join(self.CMAKE_CACHE_TYPES)
+                    raise ConanException(f'CMakeToolchain invalid type "{var_type}" for cache variable "{key}". Possible types: {possible_types}')
```

Comment:
```suggestion
                    raise ConanException(f'CMakeToolchain invalid type "{var_type}" for cache variable "{key}". Possible types: {self.CMAKE_CACHE_TYPES}')
```
Probably printing directly the list is good?

Reply:
Looks odd when formatting the error, I've fixed it a little bit